### PR TITLE
Clang tidy updates

### PR DESCRIPTION
**Pull request by hjmjohnson**
This is a series of code enhancements to make the code base modernized with the C++11 standard.

### CODE REVIEW

**Comment by hjmjohnson** [comment]
@malaterre Could these pull requests please be reviewed?

**Comment by malaterre** [comment]
This PR requires me to actually reject some commits in the middle of the PR. this is a lot of work, will defer for now.

Change your compilation option to use the openssl flag. thanks

File: Source/Common/gdcmBase64.cxx [hunk 1]
```diff
@@ -104,7 +104,8 @@ static std::string base64_decode(std::string const& encoded_string)
   size_t i = 0;
   size_t j = 0;
   size_t in_ = 0;
-  unsigned char char_array_4[4], char_array_3[3];
+  unsigned char char_array_4[4];
+  unsigned char char_array_3[3];
   std::string ret;
```
Comment (malaterre) [inline Source/Common/gdcmBase64.cxx:109]: why ?

File: Source/Common/gdcmSHA1.h [hunk 2]
```diff
@@ -37,9 +37,9 @@ public :
   SHA1(const SHA1&) = delete;
   void operator=(const SHA1&) = delete;
 
-  static bool Compute(const char *buffer, unsigned long buf_len, char digest_str[20*2+1]);
+  static bool Compute(const char *buffer, unsigned long buf_len, const char digest[20*2+1]);
```
Comment (malaterre) [inline Source/Common/gdcmSHA1.h:40]: No

File: Source/Common/gdcmSystem.cxx [hunk 3]
```diff
@@ -760,7 +760,13 @@ bool System::ParseDateTime(time_t &timep, long &milliseconds, const char date[22
...
   // instead write our own:
-  int year, mon, day, hour, min, sec, n;
+  int year;
+  int mon;
+  int day;
+  int hour;
+  int min;
+  int sec;
+  int n;
   if ((n = sscanf(date, "%4d%2d%2d%2d%2d%2d",
```
Comment (malaterre) [inline Source/Common/gdcmSystem.cxx:770]: why ?

File: Source/DataStructureAndEncodingDefinition/gdcmFile.cxx [hunk 4]
```diff
@@ -16,10 +16,6 @@
 namespace gdcm_ns
 {
 
-// Keep cstor and dstor here to keep API minimal (see dllexport issue with gdcmstrict::)
-File::File() = default;
```
Comment (malaterre) [inline Source/DataStructureAndEncodingDefinition/gdcmFile.cxx:20]: Please read the comment

File: Source/DataStructureAndEncodingDefinition/gdcmReader.h [hunk 5]
```diff
@@ -61,7 +61,7 @@ class GDCM_EXPORT Reader
 
   /// Set the filename to open. This will create a std::ifstream internally
   /// See SetStream if you are dealing with different std::istream object
-  void SetFileName(const char *filename_native);
+  void SetFileName(const char *utf8path);
```
Comment (malaterre) [inline Source/DataStructureAndEncodingDefinition/gdcmReader.h:64]: no

**Comment by malaterre** [comment]
@hjmjohnson please rebase, remove the 'auto' patch (not going to take it).

**Pull request closed without merging**